### PR TITLE
HW Inventory Update Stored Procedures

### PR DESCRIPTION
**Pull request by cmcheung-UCS**
Signed-off-by: cmcheung <chi.m.cheung@intel.com>

### CODE REVIEW

File: components.pipeline [hunk 1]
```diff
@@ -0,0 +1,66 @@
+pipeline {
```
Comment (paul-amonson) [inline components.pipeline]: Missing copyright header.

File: dai_core/src/main/java/com/intel/dai/dsapi/HWInvApi.java [hunk 2]
```diff
@@ -1,9 +1,12 @@
 package com.intel.dai.dsapi;
```
Comment (paul-amonson) [inline dai_core/src/main/java/com/intel/dai/dsapi/HWInvApi.java]: Missing copyright header.

File: dai_core/src/main/java/com/intel/dai/dsapi/HWInvLoc.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.intel.dai.dsapi;
```
Comment (paul-amonson) [inline dai_core/src/main/java/com/intel/dai/dsapi/HWInvLoc.java]: Missing copyright header.

File: dai_core/src/main/java/com/intel/dai/dsapi/HWInvTree.java [hunk 4]
```diff
@@ -4,18 +4,18 @@
 import java.util.ArrayList;
```
Comment (paul-amonson) [inline dai_core/src/main/java/com/intel/dai/dsapi/HWInvTree.java]: Missing copyright header.

File: dai_core/src/main/java/com/intel/dai/dsapi/HWInvUtil.java [hunk 5]
```diff
@@ -5,10 +5,12 @@
```
Comment (paul-amonson) [inline dai_core/src/main/java/com/intel/dai/dsapi/HWInvUtil.java]: Missing copyright header.

File: dai_core/src/main/java/com/intel/dai/dsimpl/voltdb/HWInvUtilImpl.java [hunk 6]
```diff
@@ -1,9 +1,11 @@
 package com.intel.dai.dsimpl.voltdb;
```
Comment (paul-amonson) [inline dai_core/src/main/java/com/intel/dai/dsimpl/voltdb/HWInvUtilImpl.java]: Missing copyright header.

File: dai_core/src/main/java/com/intel/dai/dsimpl/voltdb/VoltHWInvApi.java [hunk 7]
```diff
@@ -3,12 +3,14 @@
 import com.google.gson.JsonIOException;
```
Comment (paul-amonson) [inline dai_core/src/main/java/com/intel/dai/dsimpl/voltdb/VoltHWInvApi.java]: Missing copyright header.

File: dai_core/src/test/groovy/com/intel/dai/dsimpl/voltdb/HWInvUtilImplSpec.groovy [hunk 8]
```diff
@@ -13,26 +13,31 @@ class HWInvUtilImplSpec extends spock.lang.Specification {
         "rm -f build/tmp/readOnly.json build/tmp/somefile.json".execute().text
```
Comment (paul-amonson) [inline dai_core/src/test/groovy/com/intel/dai/dsimpl/voltdb/HWInvUtilImplSpec.groovy]: Missing copyright header.

File: dai_core/src/test/groovy/com/intel/dai/dsimpl/voltdb/VoltHWInvApiSpec.groovy [hunk 9]
```diff
@@ -15,17 +15,21 @@ class VoltHWInvApiSpec extends spock.lang.Specification {
         api = new VoltHWInvApi(logger, new HWInvUtilImpl(), servers)
     }
```
Comment (paul-amonson) [inline dai_core/src/test/groovy/com/intel/dai/dsimpl/voltdb/VoltHWInvApiSpec.groovy]: Missing copyright header.

File: inventory/src/main/java/com/intel/dai/inventory/api/HWInvDbClientCLI.java [hunk 10]
```diff
@@ -57,55 +65,131 @@ private int run(String[] args) {
...
+                logger.info("out: canonicalOutputFileName: %s", canonicalOutputFileName);
+                return dump(client, canonicalOutputFileName);
 
             default:
                 logger.error("Unsupported mode: %s", mode);
                 return 1;
         }
     }
+
+    private int ingest(HWInvApi client, String inputFileName, boolean update) throws InterruptedException, IOException, DataStoreException {
```
Comment (yhuan47) [inline inventory/src/main/java/com/intel/dai/inventory/api/HWInvDbClientCLI.java]: Could you add more comments? on class, method level, thanks!

**Comment by cmcheung-UCS** [comment]
This commit can be automation in our official pipeline.

**Pull request merged**